src/Npgsql/ConnectorPool.cs

```diff
@@ -125,11 +129,15 @@ internal ConnectorPool(NpgsqlConnectionStringBuilder settings, string connString
             _autoPrepare = settings.MaxAutoPrepare > 0;
             _connectionLifetime = TimeSpan.FromSeconds(settings.ConnectionLifetime);
             _connectors = new NpgsqlConnector[_max];
+            _maxExtraConnectors = settings.ExtraPoolCapacity;
 
             // TODO: Validate multiplexing options are set only when Multiplexing is on
 
             if (Settings.Multiplexing)
             {
+                if (settings.ExtraPoolCapacity > 0)
+                    throw new ArgumentException("ExtraPoolCapacity is not supported with multiplexing");
```

Comment:
I think we should just close these connections after a single multiplexing roundtrip (so after all readers are done). This feature doesn't have to be mutually exclusive with multiplexing. @roji what are your thoughts?